### PR TITLE
Update insert sort

### PR DESCRIPTION
**Pull request by shellhub**
update insert sort

### **Describe your change:**

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: sorts/insertion_sort.py [hunk 1]
```diff
@@ -24,30 +24,38 @@ def insertion_sort(collection: list) -> list:
...
-            insertion_index > 0
-            and collection[insertion_index - 1] > collection[insertion_index]
-        ):
-            collection[insertion_index], collection[insertion_index - 1] = (
-                collection[insertion_index - 1],
-                collection[insertion_index],
-            )
-            insertion_index -= 1
-
+    for i in range(1, len(collection)):
```
Comment (nomisrevol) [inline sorts/insertion_sort.py:43]: Please choose a more meaningful variable name stand for `j` and `key`.

**Review by nomisrevol — DISMISSED** [review]

File: sorts/insertion_sort.py (hunk 1 above)
Comment (shellhub) [reply sorts/insertion_sort.py:43]: Done. Please review again.

Comment (dhruvmanila) [reply sorts/insertion_sort.py:43]: As mentioned, please use a descriptive variable name.

Comment (cclauss) [reply sorts/insertion_sort.py:43]: Is there a way to change this to use `enumerate()` instead of `range(len())`? if not, nevermind.

**Comment by shellhub** [comment]
@cclauss @nomisrevol @poyea @dhruvmanila Build passed. I think this PR can be merged.

File: sorts/insertion_sort.py (hunk 1 above)
Comment (dhruvmanila) [reply sorts/insertion_sort.py:43]: Let me try:

Yeah it is possible:
```python
def i_sort(collection):
    for insert_index, insert_value in enumerate(collection[1:]):
        reference = insert_index
        while insert_index >= 0 and insert_value < collection[insert_index]:
            collection[insert_index + 1] = collection[insert_index]
            insert_index -= 1
        if insert_index != reference:
            collection[insert_index + 1] = insert_value
    return collection
```
@shellhub Can you try this out?

Comment (shellhub) [reply sorts/insertion_sort.py:43]: Yes. No problem. can you test using `timeit`. I want to know which version is faster.

Comment (dhruvmanila) [reply sorts/insertion_sort.py:43]: Sure, it seems they're equivalent.

**Comment by shellhub** [comment]
@cclauss @nomisrevol @poyea @dhruvmanila Do you have time to review this ?

**Review by dhruvmanila — APPROVED** [review]
Sorry, I got busy then. 
Looks good. Awesome work!

**Pull request merged**